File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^A-Za-z0-9]","").toLowerCase();
        int left = 0;
        int right = s.length() - 1;
        boolean isPalindrome = true;
    
        while(left <= right){
            if(s.charAt(left) == s.charAt(right)){
                left++;
                right--;
            } else {
                isPalindrome = false;
                break;
            }
        }

        return isPalindrome;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        start, end = 0, len(s) - 1

        while start < end:
            if not s[start].isalnum():
                start += 1
                continue

            if not s[end].isalnum():
                end -= 1
                continue
            
            if s[start].lower() != s[end].lower():
                return False

            start += 1
            end -= 1
            
        return True